P1_BMI_Stats.R

```diff
@@ -256,65 +252,399 @@ bmiData <- bmiData %>%
-bmiData <- mutate(bmiData, ageyr = round(1000*trunc(agemth)/12)/1000)
+# Child's age in months will lie between two ages in 
+# in the lookup table. Line below calculates the next 
+# lowest whole month and converts to years.
+bmi_data <- mutate(bmi_data, ageyr = round(1000*trunc(agemth)/12)/1000)
 
 
 
 
 # The look-up file is matched in using the lowest whole month converted to years.
 # The corresponding L,M,S are the lowest (LO) values used in the interpolation.
-bmiData <- arrange(bmiData, ageyr)
+bmi_data <- arrange(bmi_data, ageyr)
 
-grd <- readRDS(file.path(hostFolder, "ReferenceFiles", "UK1990_BMI_Growth_Reference_Data.rds"))
+grd <- readRDS(file.path(host_folder, "ReferenceFiles",
+                         "UK1990_BMI_Growth_Reference_Data.rds"))
 
 # Merge data
-bmiData <- merge(bmiData, grd, by = c("ageyr"), all.x = TRUE, all.y = TRUE)
-
-
-
-
-
-
-
+bmi_data <- bmi_data %>%
+  merge(bmi_data, grd, by = "ageyr", all.x = TRUE, all.y = TRUE)
+# Rename the Growth Reference variables for the lowest whole month
+# converted to years (LO)
+  dplyr::rename(LMLO_b = bmi_male_l, MMLO_b = bmi_male_m, SMLO_b = bmi_male_s,
+                LFLO_b = bmi_female_l,MFLO_b = bmi_female_m, SFLO_b = bmi_female_s,
+                LMLO_h = height_male_l, MMLO_h = height_male_m, SMLO_h = height_male_s,
+                LFLO_h = height_female_l, MFLO_h = height_female_m, SFLO_h = height_female_s,
+                LMLO_w = weight_male_l, MMLO_w = weight_male_m, SMLO_w = weight_male_s,
+                LFLO_w = weight_female_l, MFLO_w = weight_female_m, SFLO_w = weight_female_s,
+                agelo = ageyr)
+
+
+# Child's age in months will lie between two ages in 
+# in the lookup table. Line below calculates the next 
+# highest whole month and converts to years.
+bmi_data <- mutate(bmi_data, ageyr = round(1000*(trunc(agemth)+1)/12)/1000)
 
+# Merge data
+bmi_data <-  merge(bmi_data, grd, by = c("ageyr"), all.x = TRUE, all.y = TRUE)
+  # Rename the Growth Reference variables for the highest whole month
+  # converted to years (HI)
+  dplyr::rename(bmi_data, LMHI_b = bmi_male_l, MMHI_b = bmi_male_m, SMHI_b = bmi_male_s, 
+                LFHI_b = bmi_female_l, MFHI_b = bmi_female_m, SFHI_b = bmi_female_s,
+                LMHI_h = height_male_l, MMHI_h = height_male_m, SMHI_h = height_male_s,
+                LFHI_h = height_female_l, MFHI_h = height_female_m, SFHI_h = height_female_s,
+                LMHI_w = weight_male_l, MMHI_w = weight_male_m, SMHI_w = weight_male_s, 
+                LFHI_w = weight_female_l, MFHI_w = weight_female_m, SFHI_w = weight_female_s,
+                agehi = ageyr)
+
+# Calculate age in years to 2 decimal places for BMI, Height and Weight.
+bmi_data <- bmi_data %>%
+  mutate(ageyrs2decimal = round((agemth/12), 2))
+
+#Change merge above to alternative join so following line no needed?
+bmi_data <- subset(bmi_data, !is.na(sex))
+# Interpolation - BMI
```

Comment:
Just a side note: this whole thing can be made a lot more straightforward if before this section the male and female versions of each variable are combined. As males won't have a value for the female variables and vice versa (I'm assuming - correct me if I'm wrong), you just need one variable. That would eliminate the need for the `case_when()` functions as you can apply the same calculation with the same variables to all records. 

Hope this makes sense! Although if this code works I wouldn't fuss trying to change it for now in the interest of time. Perhaps this can be an issue for something to resolve after the RAP is completed? Happy to open the issue myself.